File: src/main/java/io/jsql/hazelcast/test/Map.java
package io.jsql.hazelcast.test;

import com.hazelcast.config.Config;
import com.hazelcast.core.Hazelcast;
import com.hazelcast.core.HazelcastInstance;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.concurrent.ConcurrentMap;

/**
 * Created by 长宏 on 2017/5/3 0003.
 */
public class Map {
    static Logger logger = LoggerFactory.getLogger("me");
    public static void main(String[] args) {
        Config config = new Config();
        HazelcastInstance h = Hazelcast.newHazelcastInstance(config);
        ConcurrentMap<String, String> map = h.getMap("mymap");
        map.put("key", "value");
        map.get("key");

        logger.info(map.get("key"));
        //Concurrent Map methods
        map.putIfAbsent("somekey", "somevalue");
        map.replace("key", "value", "newvalue");
    }

}

File: src/main/java/io/jsql/hazelcast/test/Map.kt
/*
 * Java-based distributed database like Mysql
 */

package io.jsql.hazelcast.test

import com.hazelcast.config.Config
import com.hazelcast.core.Hazelcast
import com.hazelcast.core.IMap
import org.slf4j.LoggerFactory

/**
 * Created by 长宏 on 2017/5/3 0003.
 */
object Map {
    internal var logger = LoggerFactory.getLogger("me")
    @JvmStatic
    fun main(args: Array<String>) {
        val config = Config()
        val h = Hazelcast.newHazelcastInstance(config)
        val map: IMap<String, String> = h.getMap<String, String>("mymap")
        map.put("key", "value")
        map["key"]
        logger.info(map["key"])
        //Concurrent Map methods
        map.putIfAbsent("somekey", "somevalue")
        map.replace("key", "value", "newvalue")
    }

}
